database/dialects.go

```diff
@@ -32,7 +34,47 @@ const (
 	DialectVertica Dialect = "vertica"
 )
 
-// NewStore returns a new [Store] implementation for the given dialect.
+// ParseDialect returns the corresponding [Dialect], if any, for a given string with external
+// origin. A supported [Dialect] value, like [DialectPostgres], may have multiple supported
+// aliases, e.g. "postgres" or "pgx". Use this function to ensure that a [Dialect] instance
+// passed to any function that accepts a [Dialect] is a valid [Dialect].
+func ParseDialect(s string) (d Dialect, err error) {
+	// Every non-error return value from this function must have an entry in the [NewStore] lookup map.
```

Comment:
To prevent drift, what do you think of having a registry?

```go
type spec struct {
	Dialect Dialect
	Aliases []string
	New     func() dialect.Querier
}

var registry = []dialectSpec{
	{
		Dialect: DialectPostgres,
		Aliases: []string{"postgres", "pgx"},
		New:     dialects.NewPostgres,
	},
	{
		Dialect: DialectMySQL,
		Aliases: []string{"mysql"},
		New:     dialects.NewMysql,
	},
	{
		Dialect: DialectSQLite3,
		Aliases: []string{"sqlite", "sqlite3"},
		New:     dialects.NewSqlite3,
	},
	// ...
}
```

This `registry` would be used by both `ParseDialect` and `NewStore`, so drift is less likely to occur.